### ISSUE TITLE
Reachability returns incorrect reachability status

### ISSUE THREAD

**Issue by HarshitDaftary** [ashleymills/Reachability.swift#375]
I implemented below code in my viewcontrollers. 

In view did load

```
do {
            try self.reachability = Reachability()
        } catch {
            
        }
```

in another method

```
 if self.reachability.allowsCellularConnection {
                print("internet is available")

            } else {
                print("Internet not available")
            }
```

This code works properly on first view controller. For second view controller, `self.reachability.allowsCellularConnection` always returns `true`.

**Comment by kostiag** [comment]
I second this.

**Comment by HarshitDaftary** [comment]
Yes I tried in simulator. Ok I will try on device and update you if I face the same issue.

**Comment by JamesL007** [comment]
Do you still have the same issue on the simulator? How did you solve it to be able to test on the Simulator? Thanks

**Comment by kostiag** [comment]
I do not believe you can. Simulator uses Mac’s connection so the only way
is either actual device or turn off connectivity for Mac while testing.

**Comment by JamesL007** [comment]
@kostiag Thanks for the quick reply. I don't believe we can either in simulator as it uses the mac's connection. But I wanted to make sure. I could do a quick workaround (in simulator mode) by making a request to let's say "www.google.com" or "www.apple.com" and see if I get a 200 code back? This will tell me if I'm on connected or not (only on Simulator) since the reachability doesn't seem to be working properly in simulator.

**Comment by kostiag** [comment]
Sure you could but why ? Too much work for testing purposes rather than
just using device or unplugging your machine for the test. But whatever
works for you I guess. Have not looked into this since original question.

**Comment by JamesL007** [comment]
@kostiag When I turn on/off wifi on my machine while running from the simulator. Currently, the Reachability framework detects the connection change but returns the wrong status...makes testing on the simulator difficult...especially if you don't have a device handy to run your tests. On a device, the same code works as expected and well. It's only on the simulator the status is a big wonky

**Comment by JamesL007** [comment]
@kostiag I'll give my idea a try and see if it's solves my issue. Thanks for the inputs. :)

**Comment by GergoBurai** [comment]
+1, wrong status is returned on change, when testing on simulator
